app/models/project.js

```diff
@@ -31,8 +41,80 @@ export default class ProjectModel extends Model {
 
   @hasMany('milestone', { async: false }) milestones;
 
+  // Attributes for dashboard view
+  @attr() tab;
+
+  @attr() teammemberrole;
+
   @attr() applicantteam;
 
+  @computed('milestones')
+  get publicReviewPlannedStartDate() {
+    const cbMilestone = this.milestones.find(milestone => milestone.milestone === COMMUNITY_BOARD_REFERRAL_MILESTONE_ID);
+    if (cbMilestone) {
+      return cbMilestone.plannedstartdate;
+    }
+    return null;
+  }
+
+  // These displayMilestoneActual<Start/End>
+  // date fields represent different dates from different milestones
+  // depending on `tab`.
+  // If `tab` is 'upcoming' or 'to-review', these dates are derived from
+  // the participant's (specified by `teammemberrole`) review milestone.
+  // If `tab` is 'reviewed'...
+  //   - these dates come from the current In Progress milestone
+  //   - if there is two In Progress milestones, an array of milestone dates is returned
+  @computed('tab', 'teammemberrole', 'milestones')
+  get displayMilestoneActualStartDate() {
+    if (this.tab === 'to-review') {
+      const participantMilestoneId = MILESTONE_ID_LOOKUP[this.teammemberrole];
+      const participantReviewMilestone = this.milestones.find(milestone => milestone.milestone === participantMilestoneId);
+      return participantReviewMilestone.actualstartdate;
+    }
+    if (this.tab === 'reviewed') {
+      const inProgressMilestones = this.milestones.filter(milestone => milestone.statuscode === 'In Progress');
+      return inProgressMilestones.map(milestone => ({
+        milestone: milestone.milestone,
+        actualstartdate: milestone.actualstartdate,
+      }));
+    }
+    return null;
+  }
```

Comment:
My concern with this computed is that it can return 3 possible values: null, string, or array...

What if we made new computed for a "matrix" of possible outcomes? For example, computeds: `reviewedInProgressMilestones`, `toReviewStartDates`.

The tab property (where does that come from) seems like _view_ logic (although that state is derived from something like "LUP Status"), and therefore, doesn't belong in the model. Perhaps the template or something downstream (a component) could switch to the correct property. 

My reasoning behind this is that there's a lot of different possible outcomes for all of these computeds, you know? So something unexpected could happen.

Up to you whether you wanna change it!

Reply:
Check this out: https://en.wikipedia.org/wiki/Cyclomatic_complexity

Reply:
Ohh, I like the idea of breaking out the computeds down even more. So then each one can return at most two values (e.g. null, array). 

The tab property was part of the "new project schema" from our brainstorming with Hannah a few weeks back. I think the goal was to help relieve the frontend from some filtering logic for the tabs. 

I like this suggestion and think it's worthwhile! I'm going to open a new issue for it and tackle it separately. 

I think after writing out PR/Issue #728 , which depends on this PR, I'll have even more clear hindsight on how to best approach this.

Reply:
JK I think I can just include it in this PR 😂